### PR TITLE
Publish artifacts to maven local

### PR DESCRIPTION
**Pull request by owaiskazi19**
Signed-off-by: Owais Kazi <owaiskazi19@gmail.com>

### Description
Publish artifacts to Maven and use SDK as a library. This will help us to just import the code from SDK rather than having the whole SDK package in AD extension.

### Issues Resolved
Closes https://github.com/opensearch-project/opensearch-sdk-java/issues/98

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: build.gradle
Comment (dbwiddis) [inline build.gradle:32]: Not in this file, but I can't review a file you didn't change and the maven coordinates are a good point to make this comment.

Need to change the project name here:
https://github.com/opensearch-project/opensearch-sdk-java/blob/0727269bee55fe59a10df6ad624fa14cad9d4b1c/settings.gradle#L1

**Review by dbwiddis — CHANGES_REQUESTED** [review]

File: build.gradle
Comment (owaiskazi19) [reply build.gradle:34]: We can remove the `allProjects` here. I added it for later purposes and it's a good to have in gradle but we do need `publishing` without which we won't be able to publish to local.

**Review by saratvemulapalli — APPROVED** [review]
Looks good to me. 
Care to update our developer guide?

**Review by joshpalis — APPROVED** [review]
Other than the comments, also looks good to me!

File: build.gradle
Comment (owaiskazi19) [reply build.gradle:32]: Covered in this [PR](https://github.com/opensearch-project/opensearch-sdk-java/pull/96/files#diff-7f825392aa37acd1cee0c2e7b9bb7366ad6eac64f3e6cdd816e156bcb69d30deR1).

**Review by dbwiddis — COMMENTED** [review]
LGTM except you resolved the comment about updating `settings.gradle`but didn't change it in this PR.  Either change it here or wait until #96 is merged before merging this.

**Comment by owaiskazi19** [comment]
> LGTM except you resolved the comment about updating `settings.gradle`but didn't change it in this PR. Either change it here or wait until #96 is merged before merging this.

Pushed the change  :slightly_smiling_face:

**Review by dbwiddis — APPROVED** [review]

**Review by joshpalis — APPROVED** [review]
LGTM

**Pull request merged**
